streaming/base/dataset.py

```diff
@@ -420,6 +424,13 @@ def __init__(self,
                 raise ValueError(f'Minimum cache usage ({min_cache_usage} bytes) is larger than ' +
                                  f'the cache limit ({self.cache_limit} bytes). Please raise ' +
                                  f'`cache_limit`.')
+            if self.cache_limit < 4 * self.max_shard_size_across_all_streams:
+                raise ValueError(f'Cache limit ({self.cache_limit} bytes) is too low. ' +
+                                 f'Increase the `cache_limit` to at-least four times the ' +
+                                 f'largest shard size ({self.max_shard_size_across_all_streams} ' +
+                                 f'bytes) which includes raw (decompressed) and zip ' +
+                                 f'(compressed) file size. Recommendation is to provide a ' +
+                                 f'`cache_limit` as high as possible to avoid thrashing.')
 
```

Comment:
so to get the true max shard size, here you could add the line:

```
self.max_shard_size_across_all_streams = np.array([shard.get_max_size() for shard in self.shards])
```

and remove the other logic right?

Reply:
Done in https://github.com/mosaicml/streaming/pull/420/commits/9f7f57fbc64f9c5fa68a4a3ace1b6da70ce508d5. Also, wrap it with `max` to get the maximum value.

Reply:
yeah forgot that! ty!